### PR TITLE
Admin module-Detail user

### PR DESCRIPTION
**Pull request by dungvan**

### CODE REVIEW

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -19,4 +22,19 @@ public function index()
...
+    /**
+     * Display the specified resource.
+     *
+     * @param int $id id of user's detail
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $user = User::find($id);
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: -you should use findOrfail function.

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -19,4 +22,19 @@ public function index()
...
+        $totalOrders = Order::where(['user_id' => $id])->get();
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: -you should use with function in laravel. Get relationship in laravel.

**Review by ghost — CHANGES_REQUESTED** [review]
-check comment. please.

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -2,21 +2,40 @@
...
+     * @param int $id id of user's detail
      *
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function show($id)
     {
-        if (!isset($listUsers)) {
-            $listUsers = User::all();
+        if ($detailUser = $this->user->findOrFail($id)) {
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: what's $detailUser?

File: app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -2,21 +2,40 @@
...
+            $totalOrders = $detailUser->orders()->get();
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: refactor it. if your data has a lot of records, your sever will die.

File: composer.json [hunk 5]
```diff
@@ -7,6 +7,7 @@
     "require": {
         "php": ">=5.6.4",
         "intervention/image": "^2.4",
+        "laracasts/flash": "^3.0",
```
Comment (ghost) [inline composer.json]: while this pull request like https://github.com/AT-PHPIntership/food-market-order/pull/72/files?

**Review by ghost — CHANGES_REQUESTED** [review]
Check comment.

File: app/Http/Controllers/UserController.php (hunk 3 above)
Comment (dungvan) [reply app/Http/Controllers/UserController.php]: the $detailUser is the user response to view to show detail

File: composer.json (hunk 5 above)
Comment (dungvan) [reply composer.json]: this flash to use the message session.

File: app/Http/Controllers/UserController.php [hunk 6]
```diff
@@ -2,21 +2,39 @@
...
+     * @param int $id id of user's detail
      *
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function show($id)
     {
-        if (!isset($listUsers)) {
-            $listUsers = User::all();
+        if ($userToShow = $this->user->with('orders')->findOrFail($id)) {
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: what's $userToShow?

File: resources/views/users/show.blade.php [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+                                <tr>
+                                    <td>
+                                        <strong>
+                                            <span class="glyphicon glyphicon-bookmark text-primary"></span>
+                                            {{ __('Gender') }}
+                                        </strong>
+                                    </td>
+                                    <td class="text-primary">
+                                        {{ $user->gender == 1 ? __('Male') : __('Female') }}
```
Comment (ghost) [inline resources/views/users/show.blade.php]: what's 1?

File: resources/views/users/show.blade.php [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+                                </tr>
+                                <tr>
+                                    <td>
+                                        <strong>
+                                            <span class="glyphicon glyphicon-eye-open text-primary"></span>
+                                            {{ __('User Type') }}
+                                        </strong>
+                                    </td>
+                                    <td class="text-primary">
+                                        {{ $user->is_admin == 1? __('Admin') : __('Normal User') }}
```
Comment (ghost) [inline resources/views/users/show.blade.php]: what's 1?

**Review by ghost — CHANGES_REQUESTED** [review]
check comment.

File: resources/views/users/show.blade.php (hunk 7 above)
Comment (dungvan) [reply resources/views/users/show.blade.php]: gender is boolean type (has description in branch migration), gender = 0 -> female, gender = 1 -> male

File: resources/views/users/show.blade.php (hunk 8 above)
Comment (dungvan) [reply resources/views/users/show.blade.php]: is_admin is boolean type (has description in branch migration), is_admin = 1 -> admin user, is_admin = 0 -> normal user

File: app/Http/Controllers/UserController.php [hunk 9]
```diff
@@ -7,6 +7,7 @@
 use App\User;
 use Illuminate\Support\Facades\Auth;
 use Image;
+use App\Order;
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: Do you use Order in this file?

File: app/Http/Controllers/UserController.php [hunk 10]
```diff
@@ -23,15 +24,20 @@ public function __construct(User $user)
...
      *
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function show($id)
     {
-        $users = $this->user->paginate(10);
-
-        return view('users.index')->with('listUsers', $users);
+        if ($userToShow = $this->user->with('orders')->findOrFail($id)) {
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: what't $userToShow?

File: resources/views/users/show.blade.php (hunk 7 above)
Comment (ghost) [reply resources/views/users/show.blade.php]: you should define constant to another developer can understand it.

File: resources/views/users/show.blade.php (hunk 8 above)
Comment (ghost) [reply resources/views/users/show.blade.php]: you should define constant to another developer can understand it.

File: resources/views/users/show.blade.php [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+                                </tr>
+                                <tr>
+                                    <td>
+                                        <strong>
+                                            <span class="glyphicon glyphicon-shopping-cart text-primary"></span>
+                                            {{ __('user.total-orders') }}
+                                        </strong>
+                                    </td>
+                                    <td class="text-primary">
+                                        {{ count($user->orders()) }}
```
Comment (ghost) [inline resources/views/users/show.blade.php]: you should run count order sql when get relationship.

File: resources/views/users/show.blade.php [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+                                    <div class="box-body table-responsive no-padding">
+                                        <table class="table table-hover">
+                                            <tr>
+                                                <th>{{ __('ID') }}</th>
+                                                <th>{{ __('Ordered at') }}</th>
+                                                <th>{{ __('Status') }}</th>
+                                                <th>{{ __('Custom Address') }}</th>
+                                                <th>{{ __('Payment') }}</th>
+                                            </tr>
+                                            @foreach($user->orders as $order)
```
Comment (ghost) [inline resources/views/users/show.blade.php]: you show all orders of user?

**Review by ghost — CHANGES_REQUESTED** [review]
check comment.

File: resources/views/users/show.blade.php (hunk 11 above)
Comment (at-dungvan) [inline resources/views/users/show.blade.php]: yes! this detail user show user information and user's order history

File: resources/views/users/show.blade.php (hunk 12 above)
Comment (dungvan) [reply resources/views/users/show.blade.php]: yes! this detail user show user information and user's order history

File: resources/views/users/show.blade.php (hunk 11 above)
Comment (dungvan) [reply resources/views/users/show.blade.php]: sorry, i don't understand you mean!

**Review by ghost — CHANGES_REQUESTED** [review]
check comment.

File: app/Http/Controllers/UserController.php [hunk 13]
```diff
@@ -131,7 +132,26 @@ public function update(UserRequest $request, $id)
...
+    /**
+     * Display the specified resource.
+     *
+     * @param int $id id of user's detail
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        if ($userToShow = $this->user->with('orders')->findOrFail($id)) {
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: why do you use if statement?

**Review by ghost — CHANGES_REQUESTED** [review]
check comment

File: app/Http/Controllers/UserController.php [hunk 14]
```diff
@@ -133,7 +134,22 @@ public function update(UserRequest $request, $id)
...
+    /**
+     * Display the specified resource.
+     *
+     * @param int $id id of user's detail
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $userToShow = $this->user->with('orders')->findOrFail($id);
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: I think you should take 10~15 orders of user.

**Review by ghost — CHANGES_REQUESTED** [review]
check comment

File: app/Http/Controllers/UserController.php (hunk 14 above)
Comment (dungvan) [reply app/Http/Controllers/UserController.php]: i update to pagination, check again, thank you!

File: app/Http/Controllers/UserController.php [hunk 15]
```diff
@@ -133,7 +134,22 @@ public function update(UserRequest $request, $id)
...
+        $totalOrders = Order::where('user_id', $id)->count();
+        $listOrders = Order::where('user_id', $id)->paginate(Order::ITEMS_PER_PAGE);
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: why you get again list orders?

**Review by ghost — CHANGES_REQUESTED** [review]
check comment

**Review by ghost — APPROVED** [review]
LGTM

File: app/Http/Controllers/UserController.php [hunk 16]
```diff
@@ -133,7 +134,22 @@ public function update(UserUpdateRequest $request, $id)
...
+     * Display the specified resource.
+     *
+     * @param int $id id of user's detail
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $userToShow = $this->user->findOrFail($id);
+        $totalOrders = Order::where('user_id', $id)->count();
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: No need this variable, the total order is already in the pagination.

File: resources/views/users/show.blade.php [hunk 17]
```diff
@@ -0,0 +1,217 @@
...
+                        </div>
+                        <div class="box-body">
+                            <div class="col-md-4">
+                                @if(!isset($user->image))
+                                    <img alt="" title="" class="img-circle img-thumbnail isTooltip"
+                                         src="/images/users/default.jpg"
+                                         data-original-title="Usuario">
+                                @else
+                                    <img alt="" style="width:600px;" title="" class="img-circle img-thumbnail isTooltip"
+                                         src="/images/users/{{ $user->image }}" data-original-title="Usuario">
```
Comment (ghost) [inline resources/views/users/show.blade.php]: use `asset`

File: resources/views/users/show.blade.php [hunk 18]
```diff
@@ -0,0 +1,217 @@
...
+                                        <th>{{ __('Ordered at') }}</th>
+                                        <th>{{ __('Status') }}</th>
+                                        <th>{{ __('Custom Address') }}</th>
+                                        <th>{{ __('Payment') }}</th>
+                                    </tr>
+                                    @foreach($orders as $order)
+                                        <tr>
+                                            <td>{{ $order->id }}</td>
+                                            <td>{{ $order->updated_at }}</td>
+                                            <td>@if($order->status == 0)
```
Comment (ghost) [inline resources/views/users/show.blade.php]: please don't hardcode

**Review by ghost — CHANGES_REQUESTED** [review]
pls check my cmt

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**